core/hw/sh4/dyna/blockmanager.cpp

```diff
@@ -143,13 +99,13 @@ RuntimeBlockInfo* bm_GetStaleBlock(void* dynarec_code)
 
 void bm_AddBlock(RuntimeBlockInfo* blk)
 {
-	all_blocks.push_back(blk);
-	if (blkmap.find(blk)!=blkmap.end())
-	{
-		printf("DUP: %08X %p %08X %p\n", (*blkmap.find(blk))->addr,(*blkmap.find(blk))->code,blk->addr,blk->code);
+	auto iter = blkmap.find((void*)blk->code);
+	if (iter != blkmap.end()) {
+		printf("DUP: %08X %p %08X %p\n", iter->second->addr, iter->second->code, blk->addr, blk->code);
 		verify(false);
 	}
-	blkmap.insert(blk);
+	blkmap[(void*)blk->code] = blk;
+	all_blocks.push_back(blk);
```

Comment:
I'm not sure what the all_blocks deal is. I think we recycle blocks that are deleted and all that, instead of allocating new ones.
IMHO we should just delete all of this and if we want smarter allocation, create a pool class and use it. What do you say?

Reply:
All blocks was for GC dead blocks and PGO. The BM as is in reicast can’t forget blocks one-by-one, but that’s more of a limitation than a feature. 

I think revamping the BM is a great idea, as lon as we keep memory use in check.

We also need to backport the ndc page locking features for the JIT